src/main/java/com/kryptnostic/api/v1/sharing/IndexPair.java

```diff
@@ -0,0 +1,29 @@
+package com.kryptnostic.api.v1.sharing;
+
+import com.kryptnostic.krypto.engine.KryptnosticEngine;
+
+public final class IndexPair {
+    private IndexPair( byte[] objectSearchKey, byte[] objectAddressMatrix ) {
+        this.objectSearchKey = objectSearchKey;
+        this.objectAddressMatrix = objectAddressMatrix;
+    }
+
+    private final byte[] objectSearchKey;
+    private final byte[] objectAddressMatrix;
+
+    public static IndexPair newFromKryptnosticEngine( KryptnosticEngine engine ) {
+        return new IndexPair( engine.getObjectSearchKey(), engine.getObjectAddressMatrix() );
+    }
+
+    public byte[] getObjectSearchKey() {
+        return objectSearchKey;
+    }
+
+    public byte[] getObjectAddressMatrix() {
+        return objectAddressMatrix;
+    }
+    
+    public byte[] computeSharingPair( KryptnosticEngine engine ) {
```

Comment:
this method is really confusing, and it doesn't feel like it belongs here. it seems like should be a method on `KryptnosticEngine` instead.

you pass in the engine, then invoke a method on the engine, then invoke a method on the engine again with values from this class, which you already have access to with the getters on this class.

wouldn't this make more sense... `engine.computeSharingPair(indexPair.getObjectSearchKey(), indexPair.getObjectAddressMatrix()`